### PR TITLE
ESLint + prettier update

### PR DESCRIPTION
**Pull request by jubeira**
# Description

Follow up on dependencies update #1985.

This PR excludes solidity linters to keep the scope bounded. Some Typescript changes are included to comply with the new versions (namely indentation, and precision loss errors).

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [x] Dependency changes
- [x] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [ ] Complex code has been commented, including external interfaces --> N/A
- [ ] Tests are included for all code paths --> N/A
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

See #1957.

### CODE REVIEW

**Comment by TomAFrench** [comment]
https://github.com/protofire/solhint is a solidity linter though? Shouldn't we remove it from this PR to go in another?

**Comment by jubeira** [comment]
> https://github.com/protofire/solhint is a solidity linter though? Shouldn't we remove it from this PR to go in another?

Changing `solhint` by itself doesn't break the CI (doesn't force any solidity changes) but yes, it's probably a good idea to leave it out anyways.

File: pkg/pool-utils/test/ExternalFees.test.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ describe('ExternalFees', function () {
         expect(await mock.bptForPoolOwnershipPercentage(1, FP_100_PCT.sub(1))).to.be.eq(fp(1).sub(1));
 
         expect(await mock.bptForPoolOwnershipPercentage(fp(1), fp(0.5))).to.be.eq(fp(1));
-        expect(await mock.bptForPoolOwnershipPercentage(fp(1), fp(0.25))).to.be.almostEqual(fp(0.333333333333333333));
+        expect(await mock.bptForPoolOwnershipPercentage(fp(1), fp(0.25))).to.be.almostEqual(fp(0.3333333333333333));
```
Comment (TomAFrench) [inline pkg/pool-utils/test/ExternalFees.test.ts:37]: Was this an automated change? This seems odd.

Maybe to do with JS float precision?

**Review by TomAFrench — APPROVED** [review]
Changes look good generally. There's one which is a bit odd where we're removing a couple of decimal places.

File: pkg/pool-utils/test/ExternalFees.test.ts (hunk 1 above)
Comment (TomAFrench) [reply pkg/pool-utils/test/ExternalFees.test.ts:37]: image
Looks like it is. JS seems to trim this value automatically.

Comment (jubeira) [reply pkg/pool-utils/test/ExternalFees.test.ts:37]: Yes, the linter issued the precision error with the previous amount of decimals, so I trimmed it down until the error was gone.

The result shouldn't be affected by this change

**Pull request merged**